models/users.js

```diff
@@ -292,6 +295,64 @@ const fetchUserSkills = async (id) => {
   }
 };
 
+/**
+ * Fetches user data based on the filter query
+ *
+ * @param {Object} query - Object with query parameters
+ * @param {Array} query.levelId - Array of levelIds to filter the users on
+ * @param {Array} query.levelName - Array of levelNames to filter the users on
+ * @param {Array} query.levelNumber - Array of levelNumbers to filter the users on
+ * @param {Array} query.tagId - Array of tagIds to filter the users on
+ * @param {Array} query.state - Array of states to filter the users on
+ * @return {Promise<Array>} - Array of user documents that match the filter criteria
+ */
+
+const getUsersBasedOnFilter = async (query) => {
+  const allQueryKeys = Object.keys(query);
```

Comment:
can't we run Object.entries() once and save it to use at other places.

Reply:
again after running the object.entries() we would need to extract the `keys` from that entries as the `arraysHaveCommonItem` function only excepts an array of strings .